src/palette.c

```diff
@@ -483,7 +483,9 @@ static u8 UpdateNormalPaletteFade(void)
             }
         }
 
-        return gPaletteFade.active;
+        // gPaletteFade.active cannot change since the last time it was checked. So this
+        // is equivalent to `return PALETTE_FADE_STATUS_ACTIVE;`
+        return gPaletteFade.active ? PALETTE_FADE_STATUS_ACTIVE : PALETTE_FADE_STATUS_DONE;
```

Comment:
Why the need to complicate this with a ternary? Leaving it as it is is fine because it can never be anything other than 0 or 1 anyway.